### DISCUSSION TITLE
Do-it-yourself PowerShell deployment fails right at the start

### DISCUSSION THREAD

**Discussion by matthewjlevy** [Azure/Enterprise-Scale#795] [Q&A] [answered]
Hi all, I am trying to test deploying ESLZ using the "Do-it-yourself" AZ PowerShell commands. I have populated the variables as instructed, but the first commandlet (New-AzManagementGroupDeployment) doesn't support the parameter "topLevelManagementGroupPrefix". Is there something I am missing? Perhaps AZ PowerShell module outdated or something?

> New-AzManagementGroupDeployment -Name $DeploymentName `
>                                 -ManagementGroupId $TenantRootGroupId `
>                                 -Location $Location `
>                                 -TemplateFile .\eslzArm\managementGroupTemplates\mgmtGroupStructure\mgmtGroups.json `
>                                 -topLevelManagementGroupPrefix $ESLZPrefix `
>                                 -Verbose
> New-AzManagementGroupDeployment : A parameter cannot be found that matches parameter name 'topLevelManagementGroupPrefix'.
> At line:5 char:33
> + ...                                -topLevelManagementGroupPrefix $ESLZPr ...
> +                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>     + CategoryInfo          : InvalidArgument: (:) [New-AzManagementGroupDeployment], ParameterBindingException
>     + FullyQualifiedErrorId : NamedParameterNotFound,Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation.NewAzureManagementGroupDeploymentCmdlet

I am using VSCode and the following versions of PowerShell and Az respectively:
![image](https://user-images.githubusercontent.com/27995471/132496697-12bb1806-4405-4a9f-83bf-591304b2c53c.png)

**Comment by jtracey93** [comment, marked as answer]
Looks like you are missing a backtick on the first and second lines?

Try: 
```powershell
New-AzManagementGroupDeployment -Name $DeploymentName -ManagementGroupId $TenantRootGroupId `
-Location $Location -TemplateFile .\eslzArm\managementGroupTemplates\mgmtGroupStructure\mgmtGroups.json `
-topLevelManagementGroupPrefix $ESLZPrefix `
-Verbose
```

Could you also try updating the Az PowerShell module to the latest and then re-launch the PowerShell session. The command to update is: 
```powershell
Update-Module -Name 'Az' -Force
```

**Reply by matthewjlevy** [reply]
Thanks @jtracey93 for the response. The missing backticks in lines 1 and 2 are because I was struggling with the github markdown (ctrl+e) so may have manually removed them before my post, but I copied the snippet verbatim from the [Readme.md](https://github.com/Azure/Enterprise-Scale/blob/22c594cd6a593bd7a15a96b1c1781965939134f4/eslzArm/README.md#L38-L43)

As for the Module version, I have run `Update-Module -Name 'Az' -Force`
and it ran without errors, but I still seem to have Az.resources v2.0.1

I am now trying to get PowerShell v7 installed and then configure VSCode to use v7 as the integrated shell. Fun and games.

**Reply by jtracey93** [reply]
Keep us posted. You generally need to close/terminate the powershell sessions before the upgrade shows as effective. But PowerShell 7 is a good idea anyway.

Keep us posted

**Reply by matthewjlevy** [reply]
Updated to PowerShell v7 restarted VSCode, Then updated the AZ module, restarted VSCode and now the commandlet is working:

New-AzManagementGroupDeployment -Name $DeploymentName `
>>                                 -ManagementGroupId $TenantRootGroupId `
>>                                 -Location $Location `
>>                                 -TemplateFile .\eslzArm\managementGroupTemplates\mgmtGroupStructure\mgmtGroups.json `
>>                                 -topLevelManagementGroupPrefix $ESLZPrefix `
>>                                 -Verbose
VERBOSE: 
VERBOSE: 16:27:13 - Template is valid.

**Reply by jtracey93** [reply]
Awesome thanks for letting us know
